src/java/org/apache/cassandra/db/virtual/SettingsTable.java

```diff
@@ -59,102 +56,83 @@ final class SettingsTable extends AbstractVirtualTable
                            .addPartitionKeyColumn(NAME, UTF8Type.instance)
                            .addRegularColumn(VALUE, UTF8Type.instance)
                            .build());
-        this.config = config;
+        registerConverters();
+        this.configurationRegistry = configurationRegistry;
     }
 
-    @Override
-    public DataSet data(DecoratedKey partitionKey)
+    private void registerConverters()
     {
-        SimpleDataSet result = new SimpleDataSet(metadata());
-        String name = UTF8Type.instance.compose(partitionKey.getKey());
-        if (BACKWARDS_COMPATABLE_NAMES.containsKey(name))
-            ClientWarn.instance.warn("key '" + name + "' is deprecated; should switch to '" + BACKWARDS_COMPATABLE_NAMES.get(name) + "'");
-        if (PROPERTIES.containsKey(name))
-            result.row(name).column(VALUE, getValue(PROPERTIES.get(name)));
-        return result;
+        converterRegistry.put(Boolean.class, CassandraRelevantProperties.BOOLEAN_CONVERTER);
+        converterRegistry.put(boolean.class, CassandraRelevantProperties.BOOLEAN_CONVERTER);
+        converterRegistry.put(Integer.class, CassandraRelevantProperties.INTEGER_CONVERTER);
+        converterRegistry.put(int.class, CassandraRelevantProperties.INTEGER_CONVERTER);
+        converterRegistry.put(DurationSpec.LongMillisecondsBound.class,
+                              (PropertyConverter<DurationSpec.LongMillisecondsBound>) value -> DurationSpec.from(value, DurationSpec.LongMillisecondsBound.class));
     }
 
-    @Override
-    public DataSet data()
+    /**
+     * Setter for the property.
+     * @param name the name of the property.
+     * @param value the string representation of the value of the property to set.
+     */
+    private void setProperty(String name, String value)
     {
-        SimpleDataSet result = new SimpleDataSet(metadata());
-        for (Map.Entry<String, Property> e : PROPERTIES.entrySet())
-            result.row(e.getKey()).column(VALUE, getValue(e.getValue()));
-        return result;
+        Class<?> setterType = configurationRegistry.getPropertyType(name);
+        PropertyConverter<?> converter = converterRegistry.get(setterType);
+        if (converter == null)
+            throw invalidRequest("Unknown converter for property with name '%s' and type '%s'", name, setterType);
+
+        configurationRegistry.update(name, value == null ? null : converter.convert(value));
     }
 
-    private String getValue(Property prop)
+    private static @Nullable String getProperty(ConfigurationRegistry registry, String name)
     {
-        Object value = prop.get(config);
+        Object value = registry.get(name);
         return value == null ? null : value.toString();
```

Comment:
an issue this patch doesn't look to address is that the `input` and `output` may not match.

For basic types like int/boolean this is fine as they are simple, but cases like `hinted_handoff_disabled_datacenters` are not correct as that's java collection toString, which we won't have a property converter for...  it's also weird for users to generate java's collection toString.  There are many types that fall into this bucket.

You *should* have a property that w/e is returned from this table is accepted by this table; which is not true with this patch.